src/io/MuxReader.hh

```diff
@@ -96,13 +96,13 @@ public:
         }
         void reserve(size_t new_reserved) {
             if (new_reserved > mReserved) {
-                mReserved = new_reserved;
-                uint8_t *new_begin = (uint8_t*)mAlloc.allocate(mReserved);
+                uint8_t *new_begin = (uint8_t*)mAlloc.allocate(new_reserved);
                 if (mBegin != NULL) {
                     memcpy(new_begin, mBegin, mSize);
-                    mAlloc.destroy(mBegin);
+                    mAlloc.deallocate(mBegin, mReserved);
```

Comment:
`destroy` calls a destructor. Will `deallocate` call it?

Reply:
[JpegAllocator::deallocate()](https://github.com/Bulat-Ziganshin/lepton/blob/4e3980d8d4237f2836cf5a9254908740d71fdaf3/src/io/Allocator.hh#L144) doesn't call destroy neither destructor.

[ResizableByteBuffer destructor](https://github.com/Bulat-Ziganshin/lepton/blob/4e3980d8d4237f2836cf5a9254908740d71fdaf3/src/io/MuxReader.hh#L119) calls [JpegAllocator<uint8_t>::destroy()](https://github.com/Bulat-Ziganshin/lepton/blob/4e3980d8d4237f2836cf5a9254908740d71fdaf3/src/io/Allocator.hh#L156) which executes (meaningless) uint8_t destructor. I think we should remove destroy() call since allocation/deallocation in [ResizableByteBuffer::reserve()](https://github.com/Bulat-Ziganshin/lepton/blob/4e3980d8d4237f2836cf5a9254908740d71fdaf3/src/io/MuxReader.hh#L97) never calls constructors or destructors.